bson/primitive/decimal.go

```diff
@@ -211,6 +212,47 @@ func (d Decimal128) IsZero() bool {
 	return d.h == 0 && d.l == 0
 }
 
+// MarshalJSON returns Decimal128 as a string.
+func (d Decimal128) MarshalJSON() ([]byte, error) {
+	return json.Marshal(d.String())
+}
+
+// UnmarshalJSON creates a primitive.Decimal128 from a JSON string.
```

Comment:
This no longer supports just a string. We should document the supported formats and their behaviors (i.e. the string that would be outputted by `MarshalJSON`, extjson $numberDecimal values, and the string `null`).